### PR TITLE
[ui] Reuse RunConfigDialog for Runs list

### LINKED ISSUE #18855: Make run config box bigger/expandable
### Issue from the [Dagster Slack](https://join.slack.com/t/dagster/shared_invite/enQtNjEyNjkzNTA2OTkzLTI0MzdlNjU0ODVhZjQyOTMyMGM1ZDUwZDQ1YjJmYjI3YzExZGViMDI1ZDlkNTY5OThmYWVlOWM1MWVjN2I3NjU)

This issue was generated from the slack conversation at: https://dagster.slack.com/archives/C02LJ7G0LAZ/p1703121680219909?thread_ts=1703121680.219909&cid=C02LJ7G0LAZ

---

### Conversation excerpt

U0300665S8G: Any chance the config box that appears when you view the config for a run from the Runs page could be made a bit bigger? Or maybe it could just be expandable horizontally and vertically with a click-drag on the corner? It's currently pretty painful to view configs that are large vertically and/or horizontally. Especially on wide screen monitors it feels like a big waste of space to have this tiny little box in the middle of the screen with no way to expand it
U039VKAHCAY: <@U018K0G2Y85> issue Make run config box bigger/expandable

---

#### Message from the maintainers:

Do you care about this too? Give it a :thumbsup:. We factor engagement into prioritization.

### PR DESCRIPTION
**Pull request by hellendag**
## Summary & Motivation

Resolves #18855.

Use the run config dialog from the Run page on the Runs list. It occupies more of the viewport, which is better for runs with large config yaml content.

## How I Tested These Changes

View a run, click to view the tags/config. Verify success.

View Runs list, use action menu to view config. Verify that the large dialog appears properly.

### CODE REVIEW

**Comment by hellendag** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #18914** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/18914?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/18914?utm_source=stack-comment).

File: js_modules/dagster-ui/packages/ui-core/src/runs/RunHeaderActions.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
+import {useMutation} from '@apollo/client';
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/runs/RunHeaderActions.tsx:1]: Basically this entire file came from the old `RunConfigDialog.tsx`. I assume that file has grown in purpose over time to do more than show a run config dialog, but we never renamed the component or the dialog. This file contains the group of actions, and makes use of the actual `RunConfigDialog` component.

File: js_modules/dagster-ui/packages/ui-core/src/runs/RunRoot.tsx [hunk 2]
```diff
@@ -155,7 +155,7 @@ export const RunRoot = () => {
               </Box>
             ) : null
           }
-          right={run ? <RunConfigDialog run={run} isJob={isJob} /> : null}
+          right={run ? <RunHeaderActions run={run} isJob={isJob} /> : null}
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/runs/RunRoot.tsx:158]: See comment above regarding `RunHeaderActions.tsx`

File: js_modules/dagster-ui/packages/ui-core/src/runs/RunActionsMenu.tsx [hunk 3]
```diff
@@ -273,36 +282,14 @@ export const RunActionsMenu = React.memo(({run, onAddTag, additionalActionsForRu
...
-              });
-            }}
-          >
-            Copy config
-          </Button>
-          <Button intent="primary" onClick={closeDialogs}>
-            OK
-          </Button>
-        </DialogFooter>
-      </Dialog>
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/runs/RunActionsMenu.tsx:305]: This is the small dialog that the issue is about.

**Review by salazarm — APPROVED** [review]

**Pull request merged**